lm_eval/models/huggingface.py

```diff
@@ -0,0 +1,329 @@
+import abc
+import math
+import transformers
+import torch
+import torch.nn.functional as F
+from tqdm import tqdm
+
+from lm_eval.base import BaseLM
+from lm_eval import utils
+
+
+class HuggingFaceAutoLM(BaseLM):
```

Comment:
Should we also be storing whether the model is tf / pytorch / flax as an attribute?